docs/proposals/multiple-schedulers.md

```diff
@@ -0,0 +1,139 @@
+[#11470](https://github.com/kubernetes/kubernetes/issues/11470), the design of the multiple scheduler should be generic.
+
+## Challenges in multiple schedulers
+
+- Separating the pods
+
+    Each pod should be scheduled by only one scheduler. As for implementation, a pod should
+    have an additional field to tell by which scheduler it wants to be scheduled. Besides,
+    each scheduler, including the default one, should have a unique logic of how to add unscheduled
+    pods to its to-be-scheduled pod queue. Details will be explained in later sections.
+
+- Dealing with conflicts
+
+    Different schedulers are essentially separated processes on the system level. When all schedulers
+    try to schedule their pods on the nodes, there might be conflicts.
+
+    One example of the conflicts is resource racing: Suppose there be a `pod1` scheduled by
+    `my-scheduler` requiring 1 CPU's *request*, and a `pod2` scheduled by `kube-scheduler` (k8s native
+    scheduler, acting as default scheduler) requiring 2 CPU's *request*, while `node-a` only has 2.5
+    free CPU's, if both schedulers all try to put their pods on `node-a`, then one of them would eventually
+    fail when kubelet on `node-a` performs the create action due to insufficient CPU resources.
+
+    This conflict is complex to deal with in api-server and etcd. Our current solution is to let kubelet
+    to do the conflict check and if the conflict happens, effected pods would be put back to scheduler
+    and waiting to be scheduled again. Implementation details are in later sections.
+
+## Where to start: initial design
+
+We definitely want the multi-scheduler design to be a generic mechanism. The following lists the changes
+we want to make in the first step.
+
+- Add an annotation in pod template: `scheduler.alpha.kubernetes.io/scheduler-name`, this is used to
+separate pods between schedulers. `scheduler-name` should match one of the schedulers' `scheduler-name`
+(keep reading for more explanation)
+- Add a `scheduler-name` to each scheduler. It is done by hardcode or as command-line argument. The
+Kubernetes native scheduler (now `kube-scheduler` process) would have the name as `kube-scheduler`
+- The `scheduler-name` plays an important part in separating the pods between different schedulers
+which is done by carefully designing the condition/logic of how each scheduler selects unscheduled
+pods to add to its queue. To be specific, a scheduler can add a pod to its queue if and only if:
+    1. The pod has no host, **AND**
+    2. The `scheduler.alpha.kubernetes.io/scheduler-name` specified in the pod annotation matches
+    the scheduler name.
+
+        One exception is the default scheduler (with name `kube-scheduler`) who should be allowed
+        to add a none host pod to its queue if: a.) the pod DOES NOT contain annotation with key
+        `scheduler.alpha.kubernetes.io` **OR** b.) the pod contains annotation `scheduler.alpha.kubernetes.io/kube-scheduler`
```

Comment:
The annotation here is different than above (`scheduler.alpha.kubernetes.io/kube-scheduler` vs `scheduler.alpha.kubernetes.io/scheduler-name`).  Is that on purpose?

Why not make it simpler and say something like:

Any pod that has no `scheduler.alpha.kubernetes.io/scheduler-name` annotation is assumed to be handled by the "default scheduler".  While multi-scheduler support is in pre-release, the default scheduler is assumed to be the kubernetes built-in scheduler.  In the future it may be possible to change which scheduler is the default for a given cluster.  The kubernetes built-in scheduler will claim any pod which has no `scheduler.alpha.kubernetes.io/scheduler-name` or which  has `scheduler.alpha.kubernetes.io/scheduler-name=kube-scheduler`.